src/line.rs

```diff
@@ -259,7 +259,7 @@ fn weekdays() {
 }
 
 fn is_leap(year: i64) -> bool {
-    year % 4 == 0 && year % 100 != 0 || year % 400 == 0
+    (year & 3) == 0 && ((year % 25) != 0 || (year & 15) == 0)
```

Comment:
Could you add a comment explaining how this math maps to the leap year rules?

Also, for my edification, what about this made it faster?

Reply:
Hey, thanks for reviewing! These are all some good points. I added comments for the leap year test, and a [web link](https://stackoverflow.com/questions/3220163/how-to-find-leap-year-programatically-in-c/11595914#11595914) in the commit message for more info for those interested. Please let me know if it wouldn't be considered good etiquette to do so.

The tests get reduced and factored based on already-tested info, and we get to use fast bitwise ANDs. I removed some unneeded parentheses to make it clearer. The parentheses match the rules, otherwise the 400th year test would always be run (you can try running `if false && false || true`, and it evaluates as true!)